providers/NFTNameGeneratorProvider.tsx

```diff
@@ -0,0 +1,188 @@
+import axios from "axios"
+import { useCallback, useContext, useEffect, useMemo, useState } from "react"
+import NFTNameGeneratorContext from "./NFTNameGeneratorContext"
+import { storeBlob } from "../lib/ipfs"
+import transformTwitterHandle from "../lib/transformTwitterHandle"
+
+const textToImage = require("text-to-image")
+
+export const useNFTNameGenerator = () => useContext(NFTNameGeneratorContext)
+
+export const NFTNameGeneratorProvider = ({ children }) => {
+  const [description, setDescription] = useState<string>("")
+  const [twitterHandle, setTwitterHandle] = useState<string>("")
+  const [publicWallet, setPublicWallet] = useState<string>("")
+  const [loading, setLoading] = useState<boolean>(false)
+  const [showResults, setShowResults] = useState<boolean>(false)
+  const [name, setName] = useState<string>("")
+  const [imageURI, setImageURI] = useState<string>("")
+  const [twitterId, setTwitterId] = useState<string>("")
+  const [tokenId, setTokenId] = useState<string>("")
+  const [startConfetti, setStartConfetti] = useState<boolean>(false)
+  const [txHash, setTxHash] = useState<string>("")
+  const [haveTokenId, setHaveTokenId] = useState(false)
+
+  function dataURLtoFile(dataurl, filename) {
+    const arr = dataurl.split(",")
+    const mime = arr[0].match(/:(.*?);/)[1]
+    const bstr = atob(arr[1])
+    let n = bstr.length
+    const u8arr = new Uint8Array(n)
+
+    // eslint-disable-next-line no-plusplus
+    while (n--) {
+      u8arr[n] = bstr.charCodeAt(n)
+    }
+    return new File([u8arr], filename, { type: mime })
+  }
+
+  const checkTx = useCallback(async () => {
+    const response = await axios.get("/api/getTxLogs", {
+      params: {
+        txHash,
+      },
+    })
+    if (response) {
+      setTokenId(parseInt(response.data.logs[0].topics[3], 16).toString())
+      setHaveTokenId(true)
+    }
+  }, [txHash])
+
+  const postTweet = useCallback(
+    async (generatedName) => {
+      const response = await axios.post("/api/tweet", {
+        twitterHandle,
+        generatedName,
+      })
+      return response
+    },
+    [twitterHandle],
+  )
+
+  const getGeneratedName = useCallback(async () => {
+    const response = await axios.get("/api/randomName")
+    return response.data
+  }, [])
+
+  const generateImageURI = useCallback(
+    async (generatedName) =>
+      textToImage.generate(generatedName, {
+        debug: true,
+        fontSize: 58,
+        fontFamily: "Aileron",
+        lineHeight: 58,
```

Comment:
comment for LATER: this is how we fix Sid issue of bottom of `g` getting cut off in image, we can increase this by `~15` and should be good 👍 

- again, not for this PR, for the future, just noting